hadoop/src/main/java/org/apache/carbondata/hadoop/api/CarbonTableInputFormat.java

```diff
@@ -245,9 +245,17 @@ public CarbonTable getOrCreateCarbonTable(Configuration configuration) throws IO
     FilterResolverIntf filterInterface = carbonTable.resolveFilter(filter);
 
     // do block filtering and get split
-    List<InputSplit> splits =
-        getSplits(job, filterInterface, filteredSegmentToAccess, matchedPartitions, partitionInfo,
-            null, updateStatusManager);
+    List<InputSplit> splits = new ArrayList<>();
+    for (Segment segment : filteredSegmentToAccess) {
+      List<InputSplit> splitsPerSegment = getSplitsForSegment(job, filterInterface, segment,
```

Comment:
There is func called getSplitsOfOneSegment, can you change both func name to make it more readable